mopidy/core/history.py

```diff
@@ -28,11 +28,20 @@ def _add_track(self, track):
         name_parts = []
         if track.artists:
             name_parts.append(
-                ", ".join(artist.name for artist in track.artists)
+                ", ".join(
+                    artist.name for artist in track.artists if artist.name
+                )
             )
         if track.name is not None:
             name_parts.append(track.name)
-        name = " - ".join(name_parts)
+        are_all_artist_none = True
+        for artist in track.artists:
+            if artist.name is not None:
+                are_all_artist_none = False
```

Comment:
Insert a break statement after this line. There’s no reason to continue the loop once this is set to false.

Reply:
Done